app/js/lib/foxbox/foxbox.js

```diff
@@ -64,28 +64,27 @@ export default class Foxbox extends Service {
   init() {
     window.foxbox = this;
 
+    // No need to block the UI on the discovery process.
+    this._initDiscovery().then(() => {
+      return this[p.net].init();
+    })
+    .then(() => {
+      // Start polling.
+      this[p.settings].on('pollingEnabled', () => {
+        this.togglePolling(this[p.settings].pollingEnabled);
+      });
+      this.togglePolling(this[p.settings].pollingEnabled);
+
+      this.recipes = new Recipes({
+        settings: this[p.settings],
+        net: this[p.net]
+      });
+    });
+
     return this._initUserSession()
       .then(() => {
```

Comment:
initUserSession does two things:

1. Check if we already have a session.
2. Check if we come from a redirection.

Both things should be independent of the box discovery (and polling) process. The UI shouldn't be blocked by discovering or connecting to the box. Instead we should just inform the user about the discovery and online/offline status of the box (i.e. enabling/disabling the login button and probably later with another visual indicator on the main screen). So far we were avoiding to block the UI defaulting to localhost:3000 when the discovery process failed, but that was wrong and pretty confusing cause once we were setting that default we were stopping trying to reach out the real box url. This is specially annoying now that localhost:3000 does not work any more.